runtime-modules/council/src/lib.rs

```diff
@@ -832,14 +829,15 @@ impl<T: Trait> EnsureChecks<T> {
 
         let candidate = Candidates::<T>::get(council_user_id);
 
-        // ensure candidacy was announced in current election cycle
-        if candidate.cycle_id != CurrentAnnouncementCycleId::get() {
-            return Err(Error::NotCandidatingNow);
-        }
-
         // ensure candidacy announcing period is running now
         match Stage::<T>::get().stage {
-            CouncilStage::Announcing(_) => (),
+            CouncilStage::Announcing(_) => {
+                // ensure candidacy was announced in current election cycle
+                match Stage::<T>::get().stage {
```

Comment:
What is going on here? you are matching on the same thing `Stage::<T>::get().stage` nested? Doesn't the second math really need to be a check that the candidacy is from the same cycle?

Reply:
Probably unintentional rewrite happened when copy/pasting the snippet. Check for `AnnouncementPeriodNr` belongs there. Fixed